components/NotificationOptions.vue

```diff
@@ -0,0 +1,185 @@
+        <v-icon name="bell" scale="2" class="notification-list__icon" />
+        <b-badge v-if="unreadNotificationCount" variant="danger" class="notification-badge">
+          {{ unreadNotificationCount }}
+        </b-badge>
+        <div v-if="!smallScreen" class="nav-item__text">
+          Notifications
+        </div>
+      </div>
+    </template>
+    <b-dropdown-item class="text-right" link-class="notification-list__item">
+      <b-btn variant="white" size="sm" @click="markAllRead">
+        Mark all read
+      </b-btn>
+    </b-dropdown-item>
+    <b-dropdown-divider />
+    <b-dropdown-item v-for="notification in notifications" :key="'notification-' + notification.id" class="p-0 notpad" link-class="notification-list__item">
+      <Notification :notification="notification" class="p-0" @showModal="showAboutMe" />
+    </b-dropdown-item>
+    <infinite-loading :distance="distance" @infinite="loadMoreNotifications">
+      <span slot="no-results" />
+      <span slot="no-more" />
+      <span slot="spinner">
+        <b-img-lazy src="~/static/loader.gif" alt="Loading" />
+      </span>
+    </infinite-loading>
+  </component>
+</template>
+
+<script>
+const Notification = () => import('~/components/Notification')
+const InfiniteLoading = () => import('vue-infinite-loading')
+
+export default {
+  name: 'NotificationOptions',
+  components: {
+    InfiniteLoading,
+    Notification
+  },
+  props: {
+    distance: {
+      type: Number,
+      required: true
+    },
+    smallScreen: {
+      type: Boolean,
+      required: false,
+      default: false
+    }
+  },
+  computed: {
+    notificationType() {
+      return this.smallScreen ? 'b-dropdown' : 'b-nav-item-dropdown'
+    },
+    notifications() {
+      return this.$store.getters[
+        'notifications/getCurrentListInDescendingDateOrder'
+      ]
+    },
+    unreadNotificationCount() {
+      return this.$store.getters['notifications/getUnreadCount']
+    }
+  },
+  watch: {
+    unreadNotificationCount: function() {
+      this.$emit('unread-notification-count', this.unreadNotificationCount)
+    }
+  },
+  mounted() {
+    const me = this.$store.getters['auth/user']
+
+    if (me && me.id) {
+      // Get notifications and poll regularly for new ones.  Would be nice if this was event driven instead but requires server work.
+      this.$store.dispatch('notifications/updateNotifications')
+    }
+  },
+  methods: {
+    loadLatestNotifications() {
+      // We want to make sure we have the most up to date notifications.
+      this.complete = false
+      this.$store.dispatch('notifications/clear')
+      this.$store.dispatch('notifications/fetchNextListChunk')
+    },
+    loadMoreNotifications: function($state) {
+      const currentCount = this.notifications.length
+
+      if (this.complete) {
+        $state.complete()
+      } else {
+        this.busy = true
+        this.$store
+          .dispatch('notifications/fetchNextListChunk')
+          .then(() => {
```

Comment:
When I wrote this I was clueless about await.  While we're restructuring this code, could we rework it to use that?

Reply:
Just checked in some code to use await.  It looks much tidier.  You probably should take a proper look at this bit.  I think I've got everything but was unsure about some of the infinite-loading values.  e.g. Do we still need the busy property?